### ISSUE TITLE
Regression in Target skip behavior with 8.1.0

### ISSUE THREAD

**Issue by TobiasBreuer** [nuke-build/nuke#1416]
### Usage Information

8.1.0 / net8.0 / Windows 11

### Description

After the update to 8.1.0 we see an unexpected change in the skip behavior of our targets.
I assume that this is related to the changes regarding #1414 but not sure entirely.

### Reproduction Steps

``` csharp
class Build : NukeBuild
{
    public static int Main() => Execute<Build>(x => x.Publish);

    [Parameter("Configuration to build - Default is 'Debug' (local) or 'Release' (server)")]
    readonly Configuration Configuration = IsLocalBuild ? Configuration.Debug : Configuration.Release;

    Target Clean => _ => _
        .Before(Restore)
        .Executes(() =>
        {
        });

    Target Restore => _ => _
        .Executes(() =>
        {
        });

    Target Compile => _ => _
        .DependsOn(Restore)
        .Executes(() =>
        {
        });

    Target Publish => _ => _
        .DependsOn(Clean, Compile)
        .Triggers(CreateTag)
        .WhenSkipped(DependencyBehavior.Skip)
        .Executes(() =>
        {
        });

    Target CreateTag => _ => _
        .DependsOn(Publish)
        .OnlyWhenStatic(() => IsServerBuild)
        .Executes(() =>
        {
        });
}
```

### Expected Behavior

With Nuke 8.0.0 we see the following outcome:

![image](https://github.com/user-attachments/assets/d5a227f6-dec2-44d0-9c28-5f5b0aa8e582)

### Actual Behavior

After update to Nuke 8.1.0 the outcome is however

![image](https://github.com/user-attachments/assets/d634c667-eca3-4ca2-be8a-52ecdb289b92)

### Regression?

Yes, as mentioned above this worked in 8.0.0 as (at least we) expected.
Please note that also changing the execution and trigger definition to

``` csharp
public static int Main() => Execute<Build>(x => x.CreateTag);

// Remove trigger from publish target:
Target Publish => _ => _
    .DependsOn(Clean, Compile)
    // .Triggers(CreateTag)
    // .WhenSkipped(DependencyBehavior.Skip)
    .Executes(() =>
    {
    });
```
Does not change the outcome.

### Known Workarounds

Only option seems to be downgrade to 8.0.0 again.

I wonder if it is a mis-interpretation of the behavior on our end, but I'm very suprised by the semantic that a downstream target which is skipped due to some condition leads to the fact that earlier targets are skipped now.

### Could you help with a pull-request?

No

**Comment by TobiasBreuer** [comment]
Update regarding a possible **Workaround**  
Changing the `CreateTag` target to include `DependencyBehavior.Execute` will result in the same execution as with 8.0.0:

``` csharp
Target CreateTag => _ => _
    .DependsOn(Publish)
    .OnlyWhenStatic(() => IsServerBuild)
    .WhenSkipped(DependencyBehavior.Execute)
    .Executes(() =>
    {
    });
```
![image](https://github.com/user-attachments/assets/557bc736-20bf-4536-ac4f-54a1e8d499b9)

However, my understanding of this config was that, when target A defines `WhenSkipped(DependencyBehavior.Skip)` this will impact all targets that have `DependsOn(A)` (or all targets that are linked in A by `DependentFor(B, C)`)

But this doesn't seem to be the case - meaning the `WhenSkipped` configuration seems to affect all targets regardless of 'direction'.

Let's add another target that depends on the `CreateTag` as defined above:

``` csharp
Target Publish => _ => _
    .DependsOn(Clean, Compile)
    .Triggers(CreateTag)
    .WhenSkipped(DependencyBehavior.Skip)
    .Executes(() =>
    {
    });

Target CreateTag => _ => _
    .DependsOn(Publish)
    .Triggers(SendTeamsMessage)
    .OnlyWhenStatic(() => IsServerBuild)
    .WhenSkipped(DependencyBehavior.Skip)
    .Executes(() =>
    {
    });

Target SendTeamsMessage => _ => _
    .DependsOn(CreateTag)
    .Executes(() =>
    {
    });
```
In Nuke 8.0.0 this behaves like I described above (and expected):
![image](https://github.com/user-attachments/assets/9dcbc7b2-3141-435a-8064-fbf5de6c6dc4)

The `SendTeamsMessage` is skipped because of `CreateTag` not being executed and the behavior is set to `Skip`

Setting the behavio to `Execute` will make the `SendTeamsMessage` target to get executed as well.
So the _Behavior_ config affects the "downstream" targets.

![image](https://github.com/user-attachments/assets/bb09b2e9-9993-44d1-9213-a2dda64a4434)

However in 8.1.0 it affects all of them. 
When setting the behavior to `Execute` all other targets will be executed while setting it to `Skip` no target will be executed...

**Comment by sfwester** [comment]
This may be addressed by this similar issue which was just resolved #1415
